pkg/admissioncontroller/webhooks/admission/namespacedeletion/admission.go

```diff
@@ -109,21 +112,19 @@ func (h *handler) admitNamespace(ctx context.Context, request admission.Request)
 	// TODO: we should use a direct lookup here, as we might falsely allow the request, if our cache is
 	// out of sync and doesn't know about the project. We should use a field selector for looking up the project
 	// belonging to a given namespace.
-	project, err := common.ProjectForNamespaceWithClient(ctx, h.cacheReader, request.Name)
+	project, namespace, err := gutil.ProjectAndNamespaceFromReader(ctx, h.cacheReader, request.Name)
```

Comment:
Can we use the new field selector here now (see `TODO`)?

Reply:
I guess no, we need the `Namespace` object further down.